### PR TITLE
Add application_config_filename field to DeploymentSource

### PR DESCRIPTION
**Pull request by Warashi**
**What this PR does / why we need it**:

As title.

As a comment, this field is used to avoid reading the application config file as an application manifest, such as Kubernetes manifests.

**Which issue(s) this PR fixes**:

Part of #4980 

**Does this PR introduce a user-facing change?**: No

- **How are users affected by this change**:
- **Is this breaking change**:
- **How to migrate (if breaking change)**:

### CODE REVIEW

**Review by t-kikuc — COMMENTED** [review]
@Warashi 
[ASK] You mean `application_config_filename` is used in pipedv1, not in pipedv0, right?

**Comment by Warashi** [comment]
@t-kikuc 
Yes. Pipedv1 and v0 share the model package, and the message DeploymentSource is used only in v1.

File: pkg/model/deployment_source.proto [hunk 1]
```diff
@@ -25,4 +25,7 @@ message DeploymentSource {
   string revision = 2;
   // The configuration of the application which is specific for plugins.
   bytes application_config = 3;
+  // The filename of the application configuration file.
+  // The plugins can use this to avoid mistakenly reading this file as a manifests.
```
Comment (t-kikuc) [inline pkg/model/deployment_source.proto:29]: [nits]
```suggestion
  // The plugins can use this to avoid mistakenly reading this file as a manifest.
```

Comment (Warashi) [reply pkg/model/deployment_source.proto]: Thanks

**Review by t-kikuc — DISMISSED** [review]

**Comment by Warashi** [comment]
oops, gen/code is failing.
I'll run `make gen/code`

**Review by khanhtc1202 — APPROVED** [review]
LGTM 👍

**Review by t-kikuc — APPROVED** [review]

**Pull request merged**